File: backend/domain/entities/member.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class Member:
    id: int
    member_number: int | None
    first_name: str
    last_name: str
    nickname: str | None
    phone: str | None
    email: str
    display_name: str
    password_hash: str | None
    is_admin: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime
    last_payment: str | None = None
    gender: str | None = None

File: frontend/src/types/member.ts
export interface CurrentMember {
  readonly id: number;
  readonly member_number: number | null;
  readonly first_name: string;
  readonly last_name: string;
  readonly nickname: string | null;
  readonly phone: string | null;
  readonly email: string;
  readonly display_name: string;
  readonly is_admin: boolean;
  readonly is_active: boolean;
  readonly last_payment: string | null;
}

export const genderLabels = ["socio", "socia", "socio/a"] as const;
export type GenderLabel = (typeof genderLabels)[number];

export interface ValidateMemberResponse {
  readonly member_number: number;
  readonly first_name: string;
  readonly last_name: string;
  readonly active: boolean;
  readonly last_payment: string | null;
  readonly gender_label: GenderLabel;
}
